python/cudf/dataframe/columnops.py

```diff
@@ -235,8 +235,14 @@ def as_column(arbitrary, nan_as_null=True, dtype=None):
                           "categorical")
             data = as_column(arbitrary.dictionary_encode())
         elif isinstance(arbitrary, pa.NullArray):
-            if dtype and dtype != 'empty':
-                new_dtype = dtype
+            if dtype:
+                if type(dtype) == str:
+                    if dtype == 'empty':
+                        new_dtype = np.dtype(arbitrary.type.to_pandas_dtype())
+                    else:
+                        new_dtype = dtype
+                else:
+                    new_dtype = dtype
             else:
                 new_dtype = np.dtype(arbitrary.type.to_pandas_dtype())
```

Comment:
This seems literally redundant - the new logic is a duplicate of the original `if/else`:

Reply:
The reason is if `dtype` is a numpy dtype instead of a string, then this throws an error because it doesn't know how to compare "empty" to it.

Reply:
Propose?
```
new_dtype = dtype
if isinstance(dtype, str) and dtype == 'empty' or dtype == None:
    new_dtype = np.dtype(arbitrary.type.to_pandas_dtype())
```
It isn't _much_ better but at least it doesn't have four leaves.

Reply:
> if isinstance(dtype, str) and dtype == 'empty' or dtype == None:

Didn't realize Python won't evaluate the other conditionals here if the first condition is false. Will update accordingly using this 😄